marcelo_zoccoler/entry_user_interf4/Readme.md

```diff
@@ -0,0 +1,335 @@
+    return [ExampleQWidget, example_magic_widget]
+```
+
+Last thing is to replace `napari_experimental_provide_dock_widget` outputs by our own. In our case, it should look like this:
+
+```Python
+@napari_hook_implementation
+def napari_experimental_provide_dock_widget():
+    # you can return either a single widget, or a sequence of widgets
+    return [Qt_Designer_flood, flood_magic_factory, FunctionGui_flood]
+```
+If you now call `napari` again, your plugin, along with each of the 3 sub-menu versions, should be available and working!
+
+![](images/flood_plugin_v1.png)
+
+Obs: remember we wrote our qt_designer version to work with a specific layer name (`napari_island`). There are other ways of selecting the appropriated layer, like by order, type, etc. You can have access to available layers through the viewer instance `viewer.layers`.
+
+## Updating Github repository
+
+We have our updates locally in our separated branch. Let's push them into our Github repository. If you open Github Desktop again now, you should see all the changes we made listed in the left of the screen. Write a description to these changes (step 1.), click on 'commit to 'branch name'' (step 2.), and then publish your new branch (step 3.):
+
+![](images/flood_plugin_commit.png)
+
+After that, click on the 'Create Pull Request' button, and you will be prompted to a browser tab with your github online repository. There, click on 'Create Pull Request'. 
+
+![](images/flood_plugin_PR.png)
+
+After the page updates itself, click on 'Merge Pull Request' and confirm it if required.
+
+![](images/flood_plugin_merge.png)
+
+At the end, you should see the image below, which means our online repository is now updated with your new code!
+
+![merge_done](https://user-images.githubusercontent.com/26173597/145111544-dea73c05-a8a5-47bb-85c4-d9431ca0de0e.png)
+
+## Publishing your plugin
+Publishing the plugin on [PyPi](https://pypi.org/) may sound like a big ordeal, but is, in fact, very simple. To do so, first create an account on [PyPi](https://pypi.org/account/register/). After this is done, open an anaconda command prompt and `cd` into your project repository. 
+
+Next, you need to create the necessary packaging information from your sourcefiles. You can do so by either creating a *source distribution* of your package with 
+```bash
+python setup.py sdist
+```
+
+or a *wheel* using
+```bash
+python setup.py bdist_wheel
+```
+
+The difference between both is, in short, that the source distribution provides, as the name suggests, the soure files that can be downloaded and have to be compiled upon installation of the package. Python wheels, on the contrary, come pre-built, which leads to faster installation. This can be convenient for large packages. For a more in-depth on the advantages of either strategy, see [this blog](https://medium.com/ochrona/understanding-python-package-distribution-types-25d53308a9a). For simple projects, there is no damage in simply providing both. 
```

Comment:
```suggestion
The difference between both is, in short, that the source distribution provides, as the name suggests, the source files that can be downloaded and have to be compiled upon installation of the package. Python wheels, on the contrary, come pre-built, which leads to faster installation. This can be convenient for large packages. For a more in-depth on the advantages of either strategy, see [this blog](https://medium.com/ochrona/understanding-python-package-distribution-types-25d53308a9a). For simple projects, there is no damage in simply providing both. 
```